app/controllers/import_controller.rb

```diff
@@ -38,108 +46,139 @@ def import
 
     # Load and parse patient import excel
     begin
-      xlsx = Roo::Excelx.new(params[:file].tempfile.path, file_warning: :ignore)
-      validate_headers(format, xlsx.sheet(0).row(1))
-      raise ValidationError.new('File must contain at least one monitoree to import', 2) if xlsx.sheet(0).last_row < 2
-      raise ValidationError.new('Please limit each import to 1000 monitorees.', 1000) if xlsx.sheet(0).last_row > 1001
+      if format == :epix
+        ext = File.extname(params[:file].tempfile.path)
+        raise FileFormatError.new('File format must be .csv', ext) unless ext == '.csv'
```

Comment:
So did we used to support excel (`.xlsx`) files here and we are choosing not to now? Or has it always had to be `.csv` and we are now just checking that more clearly?

Reply:
the Epi-X format used to be excel but now it's csv. the extension for excel files was checked by `Roo` and caught in `rescue ArgumentError, NoMethodError`, but that isn't the case for csv files, so I had to create an alternative way to throw an error if the extension wasn't correct. great question!